Classes/Command/CheckLinksCommandController.php

```diff
@@ -0,0 +1,255 @@
+            && count($optionsSettings['auth']) > 1
+        ) {
+            $clientOptions[RequestOptions::AUTH] = $optionsSettings['auth'];
+        }
+
+        return $clientOptions;
+    }
+
+    /**
+     * Returns concurrency. If not found, simply returns a default value like
+     * 10 (default).
+     */
+    protected function getConcurrency(int $concurrency): int
+    {
+        if ($concurrency >= 0) {
+            return $concurrency;
+        }
+
+        if (isset($this->settings['concurrency']) && (int)$this->settings['concurrency'] >= 0) {
+            return (int)$this->settings['concurrency'];
+        }
+
+        return 10;
+    }
+
+    /**
+     * Returns true by default and can be changed by the setting ignoreRobots
+     */
+    protected function shouldIgnoreRobots(): bool
+    {
+        return !isset($this->settings['ignoreRobots']) || (bool)$this->settings['ignoreRobots'];
+    }
+
+    /**
+     * Send notification about the result of the link check run. The notification service can be configured.
+     * Default is the emailService.
+     */
+    protected function sendNotification(array $results): void
+    {
+        $notificationServiceClass = trim($this->notificationServiceClass);
+        if ($notificationServiceClass === '') {
+            $errorMessage = 'No notification service has been configured, but the notification handling is enabled';
+            throw new \InvalidArgumentException($errorMessage, 1540201992);
+        }
+
+        $minimumStatusCode = $this->settings['notifications']['minimumStatusCode'] ?? self::MIN_STATUS_CODE;
+        $arguments = [];
+        foreach ($results as $statusCode => $urls) {
+            if ($statusCode < (int)$minimumStatusCode) {
+                continue;
+            }
+
+            $arguments['result'][$statusCode] = [
+                'statusCode' => $statusCode,
+                'urls' => $urls,
+                'amount' => count($urls)
+            ];
+        }
+
+        /** @var NotificationServiceInterface $notificationService */
+        $notificationService = $this->objectManager->get($notificationServiceClass);
+        $notificationService->sendNotification($this->settings['notifications']['subject'] ?? '', $arguments);
+    }
+
+    /**
+     * Crawl for invalid links within nodes
+     *
+     * This command crawls an url for invalid internal and external links
+     *
+     * @throws IllegalObjectTypeException
+     * @throws InvalidActionNameException
+     * @throws InvalidArgumentNameException
+     * @throws InvalidArgumentTypeException
+     * @throws InvalidControllerNameException
+     * @throws InvalidQueryException
+     * @throws MissingActionNameException
+     * @throws UnresolvedDependenciesException
+     * @throws \Neos\Eel\Exception
+     * @throws \Neos\Flow\Property\Exception
+     * @throws \Neos\Flow\Security\Exception
+     * @throws \Neos\Neos\Exception
+     */
+    public function crawlNodesCommand(): void
```

Comment:
This command did not work for me. Please test with https://github.com/code-q-web-factory/LinkCheckerDemo